js/preserveData.js

```diff
@@ -0,0 +1,21 @@
+const userName = document.querySelector('#userName');
+const userEmail = document.querySelector('#email');
+const userMessage = document.querySelector('#message');
+const submissionBtn = document.querySelector('#submit');
+
+window.addEventListener('load', () => {
+  const getUserData = JSON.parse(localStorage.getItem('setUserData'));
+  userName.value = getUserData.name;
+  userEmail.value = getUserData.email;
+  userMessage.value = getUserData.msg;
+});
+
+// set userDate to localStorage
+const setUserData = { name: '', email: '', msg: '' };
+
+submissionBtn.addEventListener('click', () => {
+  setUserData.name = userName.value;
+  setUserData.email = userEmail.value;
+  setUserData.msg = userMessage.value;
+  localStorage.setItem('setUserData', JSON.stringify(setUserData));
+});
```

Comment:
- [ ] As requested by the previous reviewer kindly consider hitting the below  points as they are requirements 
-  When the user changes the content of any input field, the data is saved to the local storage.
- When the user loads the page, if there is any data in the local storage the input fields are pre-filled with this data.

 ☠️  Do not create one local storage entry per input field. just create one local storage and as long as the user updates one field the whole local storage gets updated


doing the above will make your project stand out and fulfill all the requirements